File: src/main/java/com/blogspot/captain1653/TypeWordPredicateFactory.java
package com.blogspot.captain1653;

import java.util.function.Predicate;

public class TypeWordPredicateFactory {

    public Predicate<String> create(String typeWord) {
        switch (typeWord) {
            case "noun": return line -> line.startsWith("noun");
            case "adjective": return line -> line.startsWith("adjective");
            case "adverb": return line -> line.startsWith("adverb");
            case "verb": return line -> line.startsWith("verb");
            default: return line -> true;
        }
    }

}

File: src/main/scala/com/blogspot/captain1653/dictionary/scala/TypeWordPredicateFactory.scala
package com.blogspot.captain1653.dictionary.scala

import java.util.function.Predicate

object TypeWordPredicateFactory {
  private def TYPES = List("noun", "adjective", "adverb", "verb")

  def apply(): TypeWordPredicateFactory = new TypeWordPredicateFactory(TYPES)

}

class TypeWordPredicateFactory private (private val types: List[String]) {

  def create(typeWord: String): Predicate[String] = {
    typeWord match {
      case someType if types.contains(someType) => new Predicate[String] {
        override def test(line: String): Boolean = line.startsWith(someType)
      }
      case _ => new Predicate[String] {
        override def test(t: String): Boolean = true
      }
    }
  }

}
